build.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+N_CORES=$(sysctl -n hw.ncpu) # macOS
+# TODO: get cores linux
+
+# ESP-IDF v3.x
+esp_idf_v3_build_native() {
+    cp arduinojason.mk components/ArduinoJson/component.mk
+    make all -j$N_CORES;
+    rm components/ArduinoJson/component.mk
+}
+
+# build docker v3.x
+esp_idf_v3_build_docker() {
+    N_CORES_DOCKER=$(docker run --rm espressif/idf:v3.3.4 /bin/bash -c "grep processor /proc/cpuinfo | wc -l; exit" | tail -1)
```

Comment:
I've been using 3.3.3 locally since forever. There might have been a reason I didn't use 3.3.4, but I don't remember what it was. It might have just come out after I locked all the versions in.

Reply:
I used the Espressif official download tool for VSCode to setup my environment and it didn't have v3.3.3 as an option, only had v3.3.4. If we can't make it work on 3.3.4 we can roll back to 3.3.3.

Reply:
Seems to work just fine!